Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -3155,6 +3169,62 @@ static Color parseColorFunction(CSSParserTokenRange& range, const CSSParserConte
     return color;
 }
 
+static std::optional<ColorOrUnresolvedColor> parseColorFunction(CSSParserTokenRange& range, const CSSParserContext& context)
```

Comment:
The duplication of the big switch statement for each color function seems suspicious.

Reply:
The worker specific path is an odd duck. Keeping that working requires a bit of duplication at the root.